### PR TITLE
Severin/i879 fix drawer open telemetry

### PR DESCRIPTION
**Pull request by Baron-Severin**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/telemetry/MenuInteractionMonitor.kt [hunk 1]
```diff
@@ -5,7 +5,12 @@
 package org.mozilla.focus.telemetry
 
 /**
- * TODO
+ * Generates a basic heuristic for whether or not a user is having difficulty
+ * understanding the menu overlay.
+ *
+ * If a user opens a menu and then closes it without performing any action
+ * (represented here by clicking), it is assumed that they were unable to find
```
Comment (liuche) [inline app/src/main/java/org/mozilla/focus/telemetry/MenuInteractionMonitor.kt]: nit: "by clicking select"

**Comment by Baron-Severin** [comment]
# Request for data collection review form

1) What questions will you answer with this data?

* How often do users open the menu, get confused, and then close it without finding what they were looking for?

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?

* This heuristic will help us measure the effectiveness of our UX.  If users are frequently confused, we will need to go back and redesign our menu design.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

* We could potentially survey users or hold user research sessions, but both would risk polluting the data.  Many users might not want to admit to being confused, or even misremember their own interactions.  On top of that, this would be very time consuming and expensive.

4) Can current instrumentation answer these questions?

* We have existing telemetry for this, but due to an implementation issue it provided incorrect data.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

| Event                                    | category | method     | object                   |
|------------------------------------------|----------|------------|--------------------------|
| Menu shown by user              | action   | user_show       | menu         |
| Menu hidden by user              | action   | user_hide       | menu         |
| No menu action taken              | aggregate   | no_action_taken       | menu         |


6) How long will this data be collected?  Choose one of the following:

* Permanently (Barbara Bermes)

7) What populations will you measure?

* All users

8) If this data collection is default on, what is the opt-out mechanism for users?

* Users may opt out of data collection by turning off data collection from the settings menu.

9) Please provide a general description of how you will analyze this data.

If we notice that users frequently open and close their menus without taking an action, that will cause us to revisit our menu UX in an attempt to communicate more clearly.

10) Where do you intend to share the results of your analysis?

* On re-dash and with mobile teams.

File: app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt [hunk 2]
```diff
@@ -296,7 +300,6 @@ class BrowserFragment : IWebViewLifecycleFragment(), Session.Observer {
         browserOverlay.visibility = if (toShow) View.VISIBLE else View.GONE
```
[the remark this answers is not in the document]
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt]: Whoops.  Good catch.

File: app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt [hunk 3]
```diff
@@ -126,11 +126,11 @@ class BrowserFragment : IWebViewLifecycleFragment(), Session.Observer {
             NavigationEvent.SETTINGS -> ScreenController.showSettingsScreen(fragmentManager!!)
             NavigationEvent.LOAD_URL -> {
                 (activity as MainActivity).onTextInputUrlEntered(value!!, autocompleteResult!!, UrlTextInputLocation.MENU)
-                setOverlayVisibleByUser(false)
+                setOverlayVisibility(false)
```
Comment (liuche) [inline app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt]: Nit: visibility in android isn't actually a boolean (vis, invis, gone).

**Comment by liuche** [comment]
data-review+

1) Is there or will there be **documentation** that describes the schema for the ultimate data set available publicly, complete and accurate?

Yes, listed in this review and @Baron-Severin will add it to the Telemetry wiki for Firefox TV

@Baron-Severin, since this is a correction to an existing probe, can you also remove the documentation for the probe that has been updated so it's not duplicated in the docs?

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, this is under the data controls in Firefox TV

3) If the request is for permanent data collection, is there someone who will monitor the data over time?**
Yes, @bbinto Barbara Bermes

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?  **

Type 2, user usage of menu

5) Is the data collection request for default-on or default-off?
default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?
No, instrumenting UI interactions

7) Is the data collection covered by the existing Firefox privacy notice? **If unsure: escalate to legal if:**
Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**
No, will inform future work on whether feature is used or causes confusion to users

**Comment by Baron-Severin** [comment]
@liuche Thanks for the review.  I updated the wiki.

**Pull request merged**